### ISSUE TITLE
Can I intercept audio stream using nAudio?

### ISSUE THREAD

**Issue by dstorca** [naudio/NAudio#636]
I want to intercept an audio stream coming from a voice program(e.g. skype, discord, etc.) before it comes to me(It is not necessary to intercept the voice of specific program. 
It's okay to intercept the entire system's audio.), and then i manipulate it through python(I want to make the voice clear through deep learning.), and let it come through to my speaker. Is this possible to intercept audio stream using nAudio?

In a nutshell, 
1. Is there a way to intercept the audio using nAudio? If not, is there anything I can refer to?
2. Can I output the manipulated audio using nAudio?

**Comment by markheath** [comment]
You can use WasapiLoopbackCapture to capture the entire system's audio
You can't intercept the audio coming from a single application though
You also cant modify the system audio before it is played back

**Comment by markheath** [comment]
Can now be done in NAudio 3 (currently in preview) with WasapiRecorder
